django/contrib/gis/gdal/raster/source.py

```diff
@@ -103,33 +186,71 @@ def srs(self):
         Returns the Spatial Reference used in this GDALRaster.
         """
         try:
-            wkt = capi.get_ds_projection_ref(self.ptr)
+            wkt = capi.get_ds_projection_ref(self._ptr)
+            if not wkt:
+                return None
             return SpatialReference(wkt, srs_type='wkt')
         except SRSException:
             return None
 
-    @cached_property
+    @srs.setter
+    def srs(self, value):
+        """
+        Sets the Spatial Reference used in this GDALRaster. The input can be
```

Comment:
no space for "Spatial Reference"? (or no caps if it's not meant to refer to the class)